### PR TITLE
Add pywavelets to runtime requirements in DEPENDS.txt

### PR DESCRIPTION
**Pull request by sciunto**
Can somebody explain me why dask is in requirements.txt? and in Optional Requirements in DEPENDS.txt?

### CODE REVIEW

**Comment by soupault** [comment]
:+1:

**Comment by stefanv** [comment]
Dask is a requirement for `apply_parallel`.

**Comment by sciunto** [comment]
@stefanv OK, but it shouldn't be optional in DEPENDS.txt. right?

**Comment by emmanuelle** [comment]
I'm not sure I understand the difference between a runtime requirement and an optional requirement. Is it that a runtime requirement is needed to import skimage and its submodules, while an optional requirement is just needed for some functions?

**Comment by sciunto** [comment]
To me, yes. If a dep in runtime req is not install, skimage crashes; if a dep in opt req is not install, skimage exits gracefully with an error message for any function requiring this dep.

**Comment by sciunto** [comment]
@scikit-image/packaging Can you comment on this please?

**Comment by stefanv** [comment]
A runtime dependency is needed to run scikit-image.  An optional dependency must be installed to use certain very specific functionality, e.g. dask for parallel processing, etc.

**Comment by sciunto** [comment]
OK, then I moved dask from requirements.txt. It's ready for reviews ;)

**Comment by soupault** [comment]
@sciunto do you mind adding `dask` to `optional_requirements.txt` then?

**Comment by sciunto** [comment]
@soupault Same thought at the same time :) Done.

File: optional_requirements.txt [hunk 1]
```diff
@@ -4,3 +4,4 @@ SimpleITK; python_version != '3.4' and python_version != '3.5'
 astropy
 tifffile
 imageio
+dask[array]
```
Comment (soupault) [inline optional_requirements.txt]: Why not just to copy-paste it? I don't know the history of `>=0.5` requirement, but I believe it is there for a reason. :)

**Comment by soupault** [comment]
Hmm... Actually, we need to add `dask` wheel to our rackspace or change the infrastructure for Python 3.4 in any other suitable way:
```
+pip install --retries 3 -q -r ./optional_requirements.txt --no-index --trusted-host travis-wheels.scikit-image.org --find-links=http://travis-wheels.scikit-image.org/
/home/travis/venv/lib/python3.4/site.py:165: DeprecationWarning: 'U' mode is deprecated
  f = open(fullname, "rU")
/home/travis/venv/lib/python3.4/imp.py:32: PendingDeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  PendingDeprecationWarning)
/home/travis/venv/lib/python3.4/importlib/_bootstrap.py:579: ImportWarning: Not importing directory /home/travis/venv/lib/python3.4/site-packages/mpl_toolkits: missing __init__
  _warnings.warn(msg.format(portions[0]), ImportWarning)
  Could not find a version that satisfies the requirement dask[array]>=0.5.0 (from -r ./optional_requirements.txt (line 7)) (from versions: )
No matching distribution found for dask[array]>=0.5.0 (from -r ./optional_requirements.txt (line 7))
```

**Comment by sciunto** [comment]
@soupault I don't understand what I'm supposed to do.

**Comment by soupault** [comment]
@sciunto I'd suggest to remove everything related to `dask` and file an another issue.
If you have time though, you need to upload somehow (I'm not sure who has the access to) a `dask` `.whl` to our rackspace (travis-wheels.scikit-image.org). This is neccessary due to the differences in the ways we treat `requirements` and `optional_dependencies` (see Travis configs files for more details).

**Comment by sciunto** [comment]
Ok @soupault I opened another issue.

**Comment by soupault** [comment]
@sciunto thanks! But could you keep it as a requirement in the meanwhile (as it is in `master`)? :)
I'll help with #2330 once I return from the travel.

@stefanv what is the procedure for us to add new wheels to our rackspace?

**Comment by sciunto** [comment]
@soupault Done. I also squashed the commits.

**Pull request merged**